benchmark/compare.js

```diff
@@ -1,181 +1,85 @@
 'use strict';
-var usage = 'node benchmark/compare.js ' +
-            '<node-binary1> <node-binary2> ' +
-            '[--html] [--red|-r] [--green|-g] ' +
-            '[-- <type> [testFilter]]';
 
-var show = 'both';
-var nodes = [];
-var html = false;
-var benchmarks;
+const fork = require('child_process').fork;
+const path = require('path');
+const CLI = require('./_cli.js');
+
+//
+// Parse arguments
+//
+const cli = CLI(`usage: ./node benchmark/compare.js <type> ...
+  Run each benchmark in the <type> directory many times using two diffrent
+  node versions. More than one <type> directory can be specified. The output is
+  formatted as csv, which can be processed using for example 'compare.R'.
+
+  --new    ./new-node-binary  new node binary (required)
+  --old    ./old-node-binary  old node binary (required)
+  --runs   30                 number of samples
+  --filter pattern            string to filter benchmark scripts
+  --set    variable=value     set benchmark variable (can be repeated)
+`, {
+  arrayArgs: ['set']
+});
+
+if (!cli.optional.new || !cli.optional.old) {
+  cli.abort(cli.usage);
+  return;
+}
 
-for (var i = 2; i < process.argv.length; i++) {
-  var arg = process.argv[i];
-  switch (arg) {
-    case '--red': case '-r':
-      show = show === 'green' ? 'both' : 'red';
-      break;
-    case '--green': case '-g':
-      show = show === 'red' ? 'both' : 'green';
-      break;
-    case '--html':
-      html = true;
-      break;
-    case '-h': case '-?': case '--help':
-      console.log(usage);
-      process.exit(0);
-      break;
-    case '--':
-      benchmarks = [];
-      break;
-    default:
-      if (Array.isArray(benchmarks))
-        benchmarks.push(arg);
-      else
-        nodes.push(arg);
-      break;
-  }
+const binaries = ['old', 'new'];
+const runs = cli.optional.runs ? parseInt(cli.optional.runs, 10) : 30;
+const benchmarks = cli.benchmarks();
+
+if (benchmarks.length === 0) {
+  console.error('no benchmarks found');
+  process.exit(1);
 }
 
-var start, green, red, reset, end;
-if (!html) {
-  start = '';
-  green = '\u001b[1;32m';
-  red = '\u001b[1;31m';
-  reset = '\u001b[m';
-  end = '';
-} else {
-  start = '<pre style="background-color:#333;color:#eee">';
-  green = '<span style="background-color:#0f0;color:#000">';
-  red = '<span style="background-color:#f00;color:#fff">';
-  reset = '</span>';
-  end = '</pre>';
+// Create queue from the benchmarks list such both node versions are tested
+// `runs` amount of times each.
+const queue = [];
+for (const filename of benchmarks) {
+  for (const binary of binaries) {
+    for (let iter = 0; iter < runs; iter++) {
+      queue.push({ binary, filename, iter });
```

Comment:
I’ve been trying this PR out for #7425, and I’ve seen results marked as significant (`p` of the order of 10^-10) on benchmarks that weren’t even remotely touched (with only very slight changes in measured performance, < 1 %). I suspect background noise to be the reason for that, any I’m wondering if it would help to move the `iter` loop outwards here?